pkg/api/types.go

```diff
@@ -748,30 +748,56 @@ func (p *Properties) K8sOrchestratorName() string {
 	return ""
 }
 
-func (p *Properties) getAgentVMPrefix() string {
-	return p.K8sOrchestratorName() + "-agentpool-" + p.GetClusterID() + "-"
+// GetAgentPoolIndex returns the prefix of agentpool VMs given the pool index1
+func (p *Properties) GetAgentPoolIndex(a *AgentPoolProfile) int {
```

Comment:
//TODO: add unit test for this new function

Reply:
I think a better approach would be to just pass the name instead of the entire struct pointer. Lets pass in .Name from the template file. Also, you need to fix the godoc of this method. Personally, I would just make this a private method and not have a godoc string. But If you see a utility for this as a public method, let's make sure we fix the godoc.

Reply:
So based on that decision, I would rename it either getAgentPoolIndexByName/GetAgentPoolIndexByName.